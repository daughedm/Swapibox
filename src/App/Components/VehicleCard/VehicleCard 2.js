import React from 'react';
import '../VehicleCard/VehicleCard.css';
import heart from '../../../Assets/Images/like.svg';
import heartfilled from '../../../Assets/Images/likefilled.svg';
import PropTypes from 'prop-types';

const VehicleCard = ({ Data, image, addFavorite }) => {
  return (
    <div className="card" >
      <div className={image} alt="vehicle">
        <div  className="favorite" 
              onClick={(event) => addFavorite(Data)}>
          <img className="fav-icon" src={heart} alt="favorite" />
        </div>
        
      </div>
      <h3 className="card-header">{Data.name}</h3>
      <h5 className="sub-categories top">Model: 
        <span className="normal">{Data.model}</span>
      </h5>
      <h5 className="sub-categories">Class: 
        <span className="normal">{Data.vehicle_class}</span>
      </h5>
      <h5 className="sub-categories">Passengers: 
        <span className="normal">{Data.passengers}</span>
      </h5>
    </div>
  )
}

VehicleCard.propTypes = {
  Data: PropTypes.object,
  image: PropTypes.string,
  addFavorite: PropTypes.func
};

export default VehicleCard;