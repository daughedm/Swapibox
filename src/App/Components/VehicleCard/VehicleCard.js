import React from 'react';
import '../VehicleCard/VehicleCard.css';

const VehicleCard = ({ Data, image }) => {
  return (
    <div className="card" >
      <div className={image} alt="luke" />
      <h3>{Data.name}</h3>
      <h5>Model: {Data.model}</h5>
      <h5>Class: {Data.vehicle_class}</h5>
      <h5>Passengers: {Data.passengers}</h5>
    </div>
  )
}

export default VehicleCard;