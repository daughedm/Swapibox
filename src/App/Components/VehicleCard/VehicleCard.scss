.card {
  background-color: white;
  width: 250px;
  height: 350px;
  padding: 15px;
  margin: 15px;
  overflow: auto;
}

.card-header {
  font-family: 'Ubuntu', sans-serif;
  font-weight: 700;
  font-size: 1.5rem;
  color: rgb(92, 92, 92);
  border-bottom: 2px solid $sw-yellow;
  margin: 0;
  padding: 15px 0 15px 0;
}

.sub-categories {
  font-size: 1rem;
  font-weight: 400;
  color: rgb(92, 92, 92);
  margin: 0;
  padding: 10px 0 10px 0;
}

.normal {
  font-weight: 300;
  color: rgb(92, 92, 92);
}

.top {
  padding: 20px 0 10px 0;
}

.vehicle0 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/sandcrawler.jpg);
  background-size: cover;
}

.vehicle1 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/skyhopper.jpg);
  background-size: cover;
}

.vehicle2 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/landspeeder.jpg);
  background-size: cover;
}

.vehicle3 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/tie-fighter.jpg);
  background-size: cover;
}

.vehicle4 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Snowspeeder.jpg);
  background-size: cover;
}

.vehicle5 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/tie-bomber.jpg);
  background-size: cover;
}

.vehicle6 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/ATAT.jpg);
  background-size: cover;
}

.vehicle7 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/ATST.jpg);
  background-size: cover;
}

.vehicle8 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/twinpod.jpg);
  background-size: cover;
}

.vehicle9 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/sailbarge.jpg);
  background-size: cover;
}