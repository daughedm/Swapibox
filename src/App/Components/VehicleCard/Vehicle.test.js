import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import VehicleCard from './VehicleCard';

describe('VehicleCard', () => {
  let wrapper;
  it('should match snapshot', () => {
    const vehicle = {
      "name": "Sand Crawler",
      "model": "Digger Crawler",
      "vehicleClass": "wheeled",
      "passengers": 30
    }
    const addFavorite = jest.fn();
    wrapper = shallow(<VehicleCard item={vehicle}
      Data="vehicle"
      addFavorite={addFavorite} />)

    expect(wrapper).toMatchSnapshot();
  })
})