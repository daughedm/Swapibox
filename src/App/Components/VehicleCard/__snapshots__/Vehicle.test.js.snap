// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VehicleCard
    Data="vehicle"
    addFavorite={[Function]}
    item={
        Object {
            "model": "Digger Crawler",
            "name": "Sand Crawler",
            "passengers": 30,
            "vehicleClass": "wheeled",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          alt="vehicle"
          className={undefined}
>
          <div
                    className="favorite"
                    onClick={[Function]}
          >
                    <img
                              alt="favorite"
                              className="fav-icon"
                              src="like.svg"
                    />
          </div>
</div>,
        <h3
          className="card-header"
/>,
        <h5
          className="sub-categories top"
>
          Model:
          <span
                    className="normal"
          >
                     
          </span>
</h5>,
        <h5
          className="sub-categories"
>
          Class:
          <span
                    className="normal"
          >
                     
          </span>
</h5>,
        <h5
          className="sub-categories"
>
          Passengers:
          <span
                    className="normal"
          >
                     
          </span>
</h5>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "vehicle",
          "children": <div
            className="favorite"
            onClick={[Function]}
>
            <img
                        alt="favorite"
                        className="fav-icon"
                        src="like.svg"
            />
</div>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="favorite"
              className="fav-icon"
              src="like.svg"
/>,
            "className": "favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "favorite",
              "className": "fav-icon",
              "src": "like.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "card-header",
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Model:",
            <span
              className="normal"
>
               
</span>,
          ],
          "className": "sub-categories top",
        },
        "ref": null,
        "rendered": Array [
          "Model:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                undefined,
              ],
              "className": "normal",
            },
            "ref": null,
            "rendered": Array [
              " ",
              undefined,
            ],
            "type": "span",
          },
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Class:",
            <span
              className="normal"
>
               
</span>,
          ],
          "className": "sub-categories",
        },
        "ref": null,
        "rendered": Array [
          "Class:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                undefined,
              ],
              "className": "normal",
            },
            "ref": null,
            "rendered": Array [
              " ",
              undefined,
            ],
            "type": "span",
          },
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Passengers:",
            <span
              className="normal"
>
               
</span>,
          ],
          "className": "sub-categories",
        },
        "ref": null,
        "rendered": Array [
          "Passengers:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                undefined,
              ],
              "className": "normal",
            },
            "ref": null,
            "rendered": Array [
              " ",
              undefined,
            ],
            "type": "span",
          },
        ],
        "type": "h5",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            alt="vehicle"
            className={undefined}
>
            <div
                        className="favorite"
                        onClick={[Function]}
            >
                        <img
                                    alt="favorite"
                                    className="fav-icon"
                                    src="like.svg"
                        />
            </div>
</div>,
          <h3
            className="card-header"
/>,
          <h5
            className="sub-categories top"
>
            Model:
            <span
                        className="normal"
            >
                         
            </span>
</h5>,
          <h5
            className="sub-categories"
>
            Class:
            <span
                        className="normal"
            >
                         
            </span>
</h5>,
          <h5
            className="sub-categories"
>
            Passengers:
            <span
                        className="normal"
            >
                         
            </span>
</h5>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "vehicle",
            "children": <div
              className="favorite"
              onClick={[Function]}
>
              <img
                            alt="favorite"
                            className="fav-icon"
                            src="like.svg"
              />
</div>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="favorite"
                className="fav-icon"
                src="like.svg"
/>,
              "className": "favorite",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "favorite",
                "className": "fav-icon",
                "src": "like.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "card-header",
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Model:",
              <span
                className="normal"
>
                 
</span>,
            ],
            "className": "sub-categories top",
          },
          "ref": null,
          "rendered": Array [
            "Model:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  undefined,
                ],
                "className": "normal",
              },
              "ref": null,
              "rendered": Array [
                " ",
                undefined,
              ],
              "type": "span",
            },
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Class:",
              <span
                className="normal"
>
                 
</span>,
            ],
            "className": "sub-categories",
          },
          "ref": null,
          "rendered": Array [
            "Class:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  undefined,
                ],
                "className": "normal",
              },
              "ref": null,
              "rendered": Array [
                " ",
                undefined,
              ],
              "type": "span",
            },
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Passengers:",
              <span
                className="normal"
>
                 
</span>,
            ],
            "className": "sub-categories",
          },
          "ref": null,
          "rendered": Array [
            "Passengers:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  undefined,
                ],
                "className": "normal",
              },
              "ref": null,
              "rendered": Array [
                " ",
                undefined,
              ],
              "type": "span",
            },
          ],
          "type": "h5",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
